internal/docker/client.go

```diff
@@ -16,7 +16,7 @@ var instanceErr error
 var instance *client.Client
 var once sync.Once
 
-func GetClient() (*client.Client, error) {
```

Comment:
nit/question: do we want a convention that distinguishes getting a singleton vs creating a new instance?

Reply:
That's a good question. I could see the motivation for `GetClient` as the name since the singleton might be reused rather than creating a new client every time.

My first thought is that the singleton may not be adding much here, and could potentially be removed down the road. For now, I'm happy to revert this name change if necessary.

Reply:
this was not intended to be a blocking comment, so feel free to leave or change the name. 

I was highlighting the behavior difference compared to a typical constructor and getting a singleton while also asking if want to distinct these things. Agreed on the questionable use of singleton, could probably be removed with no consequence.

Reply:
Okay, awesome. I'll revert the name change for now because I think `NewClient` is masking a current reality that a new struct may not actually be created. I'll think more on the singleton topic for later.